deploy/determined_deploy/aws/templates/simple.yaml

```diff
@@ -415,7 +438,46 @@ Resources:
             Content-Disposition: attachment; filename="userdata.txt"
 
             #!/bin/bash
+            set -x
             mkdir -p /usr/local/determined/etc
+
+            scheme=http
+            port=8080
+            if [ ${EnableAutoTLS} = True ]; then
+              scheme=https
+              port=8443
+              cat > /usr/local/determined/etc/ssl.conf <<EOF
+              [req]
+              default_bits  = 2048
+              distinguished_name = req_distinguished_name
+              req_extensions = req_ext
+              x509_extensions = v3_req
+              prompt = no
+
+              [req_distinguished_name]
+              countryName = XX
+              stateOrProvinceName = N/A
+              localityName = N/A
+              organizationName = N/A
+              commonName = N/A
+
+              [req_ext]
+              subjectAltName = @alt_names
+
+              [v3_req]
+              subjectAltName = @alt_names
+
+              [alt_names]
+              IP.1 = $(curl http://169.254.169.254/latest/meta-data/public-ipv4)
+              IP.2 = $(curl http://169.254.169.254/latest/meta-data/local-ipv4)
```

Comment:
Is this line necessary?  It seems like if we are going to add host-name overriding in the agent, we wouldn't need this.

Reply:
Not strictly necessary (once that capability exists), but I would still consider this the more proper way of doing it. We might be able to work around it, but the certificate _should_ be valid for all addresses that the server will be using, just as a general rule.

Reply:
I suppose I was coming at it from the direction of "let minimize the number of IPs that we are authorizing" but your reasoning has convinced me.